src/utils/fillwidth.tsx

```diff
@@ -0,0 +1,77 @@
+import { reduce } from "lodash"
+
+const fillwidthDimensions = (items, containerWidth, gutter, targetHeight) => {
+  /**
+   * Scales an image object proportionally based on a direction (either -1 or 1)
+   * @param img a dimension object that references an artwork image
+   * @param dir the direction we need to scale an image, either -1 or 1
+   */
+  const resizeHeight = (img, dir) => {
+    img.width += img.width / img.height * dir
+    img.height += dir
```

Comment:
I agree, `resizeHeight` is a function private to the scope of `fillwidthDimensions`, so it’s fair to make assumptions about being able to mutate the input.